pixeltable/plan.py

```diff
@@ -737,43 +817,51 @@ def create_sample_clauses(
         group_by_clause: Optional[list[exprs.Expr]],
         order_by_clause: Optional[list[tuple[exprs.Expr, bool]]],
         limit: Optional[exprs.Expr],
-    ) -> tuple[
-        exprs.Expr,
-        Optional[list[exprs.Expr]],
-        Optional[list[tuple[exprs.Expr, bool]]],
-        Optional[exprs.Expr],
-        Optional[SampleClause],
-    ]:
-        """Implement the sample clause by creating required subclauses."""
+    ) -> SamplingClauses:
+        """tuple[
+            exprs.Expr,
+            Optional[list[exprs.Expr]],
+            Optional[list[tuple[exprs.Expr, bool]]],
+            Optional[exprs.Expr],
+            Optional[SampleClause],
+        ]:"""
+        """Construct clauses required for sampling under various conditions.
+        If there is no sampling, then return the original clauses.
+        If the sample is stratified, then return onlyt the group by clause. The rest of the
+        mechanism for stratified sampling is provided by the SampleSqlNode.
+        If the sample is non-stratified, then rewrite the query to accommodate the supplied where clause,
+        and provide the other clauses required for sampling
```

Comment:
Is there a reason to do this here, rather than in SampleSqlNode? Doing it here comes with a number of disadvantages, such as requiring more restructuring in the upcoming planner work and spreading the blast radius of this logic.